src/language-server/universal-data-definition-language-formatter.ts

```diff
@@ -21,22 +21,29 @@ export class UniversalDataDefinitionLanguageFormatter extends AbstractFormatter
         close.surround(Formatting.noSpace()).prepend(Formatting.newLine({allowMore: true})).append(Formatting.newLine({allowMore: true}));
     }
 
+    protected formatNode(node: AstNode): void {
+        const formatter = this.getNodeFormatter(node);
+        formatter.property('name').prepend(Formatting.newLine()).surround(Formatting.oneSpace({allowMore: true}))
+    }
+
     protected format(node: AstNode): void {        
         // This method is called for every AstNode in a document
         if (ast.isDataModel(node)) {
             this.formatDataModel(node);
         }
-        if(ast.isConceptualAssociation(node)){            
-            this.formatConceptualAssociation(node)
-        }
+        // test passes with this commetned check
+        // if(ast.isConceptualAssociation(node)){            
+        //     this.formatConceptualAssociation(node)
+        // }
 
         if(ast.isConceptualEntity(node)){
             this.formatConceptualEntity(node)
         }
-
-        if(ast.isConceptualBasisEntity(node)){
-            this.formatConceptualBasisEntity(node)
-        }
+        
+        // test passes with this commetned check
+        // if(ast.isConceptualBasisEntity(node)){
```

Comment:
The test you have defined only has 1 ConceptualBasisEntity. The formatting you see around it is caused by 2 things: 1) the surrounding formatting is done by `formatObj` for the containing CDM; 2) A ConceptualBasisEntity is also a ConceptualElement. So when you comment out this method, the `formatConceptualElement` will still be called.  

if you want to make sure you know the effect of a formatting method, you should have multiple instances of that type in your test. In this case, it means including multiple ConceptualBasisEntity instances. At a minimum, have 3.  It may also make sense to have instances of different types intermingled where possible just to check on the effect of formatting one type on adjacent instances of a different type. 

This brings up an important point: You can't just have `if` checks in the top level `format(AstNode)` method. If you do that, then every `if` is checked for each node and every `if` check that returns true will result in a format method being called - in the order they show up in `format(AstNode)`.  If multiple format methods are called for the same `AstNode`,not only will that be slower than necessary but also only the last one called will matter, because it will overwrite the effects of all format methods previously called for that `AstNode`.  

Instead, you need to make sure to use `if ... else if` so that only a single formatting method is called for each `AstNode`. And you need to make sure to put the formatting methods in order from most specific to most general based on the hierarchy described in the `.langium` file, because you want the first test to pass to be the most specific formatter applicable.